WordPress/Classes/ViewRelated/Reader/DetailWebView/ReaderDetailHeaderView.swift

```diff
@@ -0,0 +1,326 @@
+    func didTapBlogName()
+    func didTapMenuButton(_ sender: UIView)
+    func didTapTagButton()
+    func didTapHeaderAvatar()
+    func didTapFeaturedImage(_ sender: CachedAnimatedImageView)
+}
+
+class ReaderDetailHeaderView: UIStackView, NibLoadable {
+    @IBOutlet weak var headerView: UIView!
+    @IBOutlet weak var blavatarImageView: UIImageView!
+    @IBOutlet weak var blogURLLabel: UILabel!
+    @IBOutlet weak var blogNameButton: UIButton!
+    @IBOutlet weak var menuButton: UIButton!
+    @IBOutlet weak var featuredImageView: CachedAnimatedImageView!
+    @IBOutlet weak var featuredImageBottomPaddingView: ReaderSpacerView!
+    @IBOutlet weak var titleLabel: UILabel!
+    @IBOutlet weak var titleBottomPaddingView: UIView!
+    @IBOutlet weak var bylineView: UIView!
+    @IBOutlet weak var avatarImageView: CircularImageView!
+    @IBOutlet weak var bylineScrollView: UIScrollView!
+    @IBOutlet weak var bylineLabel: UILabel!
+    @IBOutlet weak var tagButton: UIButton!
+    @IBOutlet fileprivate var bylineGradientViews: [GradientView]!
+
+    /// The post to show details in the header
+    ///
+    private var post: ReaderPost?
+
+    /// Image loader for the featured image
+    ///
+    private lazy var featuredImageLoader: ImageLoader = {
+        // Allow for large GIFs to animate on the detail page
+        return ImageLoader(imageView: featuredImageView, gifStrategy: .largeGIFs)
+    }()
+
+    /// The user interface direction for the view's semantic content attribute.
+    ///
+    private var layoutDirection: UIUserInterfaceLayoutDirection {
+        return UIView.userInterfaceLayoutDirection(for: semanticContentAttribute)
+    }
+
+    /// Any interaction with the header is sent to the delegate
+    ///
+    var delegate: ReaderDetailHeaderViewDelegate?
+
+    func configure(for post: ReaderPost) {
+        self.post = post
+
+        configureSiteImage()
+        configureURL()
+        configureBlogName()
+        configureFeaturedImage()
+        configureTitle()
+        configureByLine()
+        configureTag()
+
+        prepareForVoiceOver()
+        prepareMenuForVoiceOver()
+        preparePostTitleForVoiceOver()
+
+        // Hide the featured image and its padding until we know there is one to load.
+        featuredImageView.isHidden = true
+        featuredImageBottomPaddingView.isHidden = true
+    }
+
+    @IBAction func didTapBlogName(_ sender: Any) {
+        delegate?.didTapBlogName()
+    }
+
+    @IBAction func didTapMenuButton(_ sender: UIButton) {
+        delegate?.didTapMenuButton(sender)
+    }
+
+    @IBAction func didTapTagButton(_ sender: Any) {
+        delegate?.didTapTagButton()
+    }
+
+    @objc func didTapHeaderAvatar(_ gesture: UITapGestureRecognizer) {
+        if gesture.state != .ended {
+            return
+        }
+
+        delegate?.didTapHeaderAvatar()
+    }
+
+    @objc func didTapFeaturedImage(_ gesture: UITapGestureRecognizer) {
+        guard gesture.state == .ended else {
```

Comment:
I know this code is copied from ReaderDetailViewController but I wonder do you know why does `gesture .state== .ended` here while `gesture.state != .ended` in `didTapHeaderAvatar` ?

Reply:
This is a very good question which I don't have an answer.

And if I change `didTapHeaderAvatar` to be like this one, it stops working!

Reply:
Interesting. I'll dive into the reason when I have more time